CMakeLists.txt

```diff
@@ -501,10 +473,8 @@ if (NOT MATH_LIBRARY)
     set(MATH_LIBRARY "")
 endif()
 
-cmake_push_check_state()
```

Comment:
what was this for?

Reply:
Just leave it for now if we don't know what it is for. It seems to unwind any variables modified in this section. It's used when `check_function_exists` is called. Let's just leave it. Try not to add changes that could possibly blow up in our face.

Reply:
Ok, left it